src/main/java/com/google/api/codegen/php/PhpDiscoveryContext.java

```diff
@@ -166,19 +171,16 @@ public String typeDefaultValue(Type type, Field field) {
         return defaultPrimitiveValue;
       } else if (kind.equals(Field.Kind.TYPE_STRING)) {
         String stringFormat = getApiaryConfig().getStringFormat(type.getName(), field.getName());
-        if (stringFormat != null) {
-          switch (stringFormat) {
-            case "date":
-              return "\'1969-12-31\'";
-            case "date-time":
-              return "\'"
-                  + new DateTime(0L).toStringRfc3339()
-                  + "\'";
-            default:
-              // Fall through
-          }
+        if (STRING_DEFAULT_MAP.containsKey(stringFormat)) {
```

Comment:
Is there a way to factor out the common cross-language code? One way would be to have the parent class implement typeDefaultValue(), and have it call methods that are overridden in the child classes to handle any initial exceptions for that language and to handle getting the correct empty string or null literal for that language. There are other ways, too.